### ISSUE TITLE
'Destination' does not appear to work as intended

### ISSUE THREAD

**Issue by jordi2k** [Jaykul/RequiredModules#2]
When passing a path (_string_) to the `-Destination` parameter it has no effect other than to create the directory.

For context; I'm calling the script as follows:
```ps
. $PSScriptRoot\resources\Install-RequiredModule.ps1 `
        -Destination "$PSScriptRoot\resources\dependencies" `
        -Path $PSScriptRoot\resources\RequiredModules.psd1 `
        -Confirm:$false `
        -TrustRegisteredRepositories
```

The directory is created, so the parameter is being read and it seems processed by `AddPSModulePath` but when processing `InstallModuleVersion` it always results in an `Install-Module` command rather than the expected `Save-Module` command.
https://github.com/Jaykul/RequiredModules/blob/fa85bd2675b72fe6ba154605d78cf8c9b92f39c7/source/Private/InstallModuleVersion.ps1#L64-L76

Unsure on the use of `-Destination:$RealDestination` since `$RealDestination` does not seem to get populated??
https://github.com/Jaykul/RequiredModules/blob/fa85bd2675b72fe6ba154605d78cf8c9b92f39c7/source/Public/Install-RequiredModule.ps1#L136

I've tried replacing with `$Destination` but this results in an error:
```ps
InstallModuleVersion : A parameter cannot be found that matches parameter name 'Scope'.
```

**Comment by wekempf** [comment]
@jordi2k I could have sworn I had this working, but like you I get the same behavior. The error you get is because `Save-Module` doesn't have a `-Scope` parameter. In the if on line 64 you'd have to remove `Scope` from `$ModuleOptions`. Alternatively, and probably better more correct, I'd remove `Scope` from the initial values set in `$ModuleOptions` (line 55) and set it in the else on line 74 instead.

@Jaykul I'd like to make a feature request here related to all of that. It doesn't make sense that you can only specify the destination when using a file. We should be able to specify it when using a hash as well.

**Comment by Jaykul** [comment]
I'm sure the `-Destination` used to work -- but obviously I don't have enough tests on this project, and I've regressed that.
I'll move this to the top of my list of things, because I've been meaning to work on this project for a couple of weeks.

**Comment by Jaykul** [comment]
Shipped in 5.0.5

**Comment by Jaykul** [comment]
@wekempf I fixed that parameterset bug too ;-)
